scripts/windows/test/Run-E2ETest.ps1

```diff
@@ -1101,22 +1101,37 @@ Function RunTransparentGatewayTest
     }
     $testCommand = AppendInstallationOption($testCommand)
     Invoke-Expression $testCommand | Out-Host
+    $result = 0;
 
-    # run the various leaf device tests
+    # run the various leaf device tests and "bitwise or" the results (-bor).
+    # This ensures that if at least one of these tests fail, then that failed result will carry through the rest of the suite and be reported as a single failure.
     $deviceId = "e2e-${ReleaseLabel}-Win-${Architecture}"
-    RunLeafDeviceTest "sas" "Mqtt" "$deviceId-mqtt-sas-noscope-leaf" $null
-    RunLeafDeviceTest "sas" "Amqp" "$deviceId-amqp-sas-noscope-leaf" $null
 
-    RunLeafDeviceTest "sas" "Mqtt" "$deviceId-mqtt-sas-inscope-leaf" $edgeDeviceId
-    RunLeafDeviceTest "sas" "Amqp" "$deviceId-amqp-sas-inscope-leaf" $edgeDeviceId
+    $result = RunLeafDeviceTest "sas" "Mqtt" "$deviceId-mqtt-sas-noscope-leaf" $null
+    $result = $result -bor 00
 
-    RunLeafDeviceTest "x509CA" "Mqtt" "$deviceId-mqtt-x509ca-inscope-leaf" $edgeDeviceId
-    RunLeafDeviceTest "x509CA" "Amqp" "$deviceId-amqp-x509ca-inscope-leaf" $edgeDeviceId
+    $result = RunLeafDeviceTest "sas" "Amqp" "$deviceId-amqp-sas-noscope-leaf" $null
```

Comment:
Should we define another variable to distinguish previous result and current test result.  If previous test is failed and returns non-zero value; if all remaining tests passed, this function will return zero and report success.  This is not expected.

Reply:
Ahh you're right! It wasn't working as I thought it did. I fixed it now and tried it out and looks tow work now.